libcudacxx/include/cuda/std/detail/libcxx/include/variant

```diff
@@ -216,31 +269,33 @@ namespace std {
 #  pragma system_header
 #endif // no system header
 
-_LIBCUDACXX_PUSH_MACROS
-#include <__undef_macros>
+#ifndef __cuda_std__
+#  include <exception>
 
-namespace std { // explicitly not using versioning namespace
+_LIBCUDACXX_BEGIN_NAMESPACE_STD_NOVERSION
 
-class _LIBCUDACXX_EXCEPTION_ABI _LIBCUDACXX_AVAILABILITY_BAD_VARIANT_ACCESS bad_variant_access : public exception {
+class _LIBCUDACXX_EXCEPTION_ABI _LIBCUDACXX_AVAILABILITY_BAD_VARIANT_ACCESS bad_variant_access : public exception
+{
 public:
-  virtual const char* what() const noexcept;
+  const char* what() const noexcept override;
 };
 
-} // namespace std
+_LIBCUDACXX_END_NAMESPACE_STD_NOVERSION
 
-_LIBCUDACXX_BEGIN_NAMESPACE_STD
+#endif // __cuda_std__
 
-#if _LIBCUDACXX_STD_VER > 14
+#if _LIBCUDACXX_STD_VER >= 14
 
-_LIBCUDACXX_NORETURN
-inline _LIBCUDACXX_INLINE_VISIBILITY
-_LIBCUDACXX_AVAILABILITY_THROW_BAD_VARIANT_ACCESS
-void __throw_bad_variant_access() {
-#ifndef _LIBCUDACXX_NO_EXCEPTIONS
-        throw bad_variant_access();
-#else
-        _CUDA_VSTD::abort();
-#endif
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+_LIBCUDACXX_NORETURN inline _LIBCUDACXX_INLINE_VISIBILITY _LIBCUDACXX_AVAILABILITY_THROW_BAD_VARIANT_ACCESS void
+__throw_bad_variant_access()
+{
+#  ifndef _LIBCUDACXX_NO_EXCEPTIONS
+  throw bad_variant_access();
+#  else
+  _LIBCUDACXX_UNREACHABLE();
```

Comment:
We should rather just call `abort` to ensure that we have deterministic failures